### DISCUSSION TITLE
Is this type of controller possible?

### DISCUSSION THREAD

**Discussion by armandhammer** [shanteacontrols/OpenDeck#63] [Q&A]
I want to design a simple controller before I do the big one with the official PCBs that I bought from you.

Here are the features I want it to have. Is all of this possible?

- USB MIDI, for receiving usb output of a keyboard midi controller (mainly notes, velocity, polyphonic aftertouch)
- 1x MIDI DIN IN
- 2x MIDI DIN OUT (to pass thru usb, DIN MIDI IN, and OpenDeck on-board controls on different channels to 2 synths)
- 1 linear pot assigned to Pitch Bend CC on MIDI Ch 1
- - can this be set to something higher than 7-bit? Like 10-bit or 14-bit?
- 1 other linear pot assigned to Mod Wheel MIDI Ch 1
- 3 buttons to switch between different presets, with LED to light up to show the active preset
- - (presets might do something like switch the PB and Mod Wheel to MIDI Ch 1 or 2, or send PB to Ch 1 and MW to Ch 2)
- 2 TRS input jacks for expression pedals, assign to CC like Filter, Resonance, Volume, Pan, etc
- 2 TS input jacks for foot switches, to send CC on/off like for Sustain or Portamento

I think the answer is Yes for everything, but I want to check!

Thanks!

**Comment by armandhammer** [comment]
Another question:

For 14-bit range on a Pitch Bend pot, would using a linear pot with a longer throw and higher resistance value give better control / outcome? I'm imagining that a 20k pot would give more analog values for the controller to interpolate into 14-bit (0 to 20,000 instead of 0 to 10,000) and a longer pot would be easier to move in small amounts and get a change in the output resistance.

Which dev board would be most appropriate for a small controller like this? I'm hoping to fit it all in a box that's around 140mm x 140mm x 30mm but maybe that's too optimistic.

**Reply by AlpesMachines** [reply]
the pot value has nothing to do with the ADC resolution of the microcontroller. 10 bits ADC allows 1024 values, 12 bits ADC 4096, etc ..
pitch bend are usually over sampled from 7 bits to 14 bits in vintage synths

**Reply by paradajz** [reply]
Pitch bend (14-bit) value on OpenDeck board is derived from 12-bit ADC value.

**Comment by paradajz** [comment]
Other than 2x MIDI out (you need to use two OpenDeck boards for this), everything else is possible, yes.

**Comment by armandhammer** [comment]
Thanks! How does Pitch Bend work with a pot? Is it set so that the middle value of the 10k pot is the center of the pitch bend, and 0 ohm is full negative pitch bend? If I use a pot with a center detent, can OpenDeck be calibrated so that the resistance at the center detent (which varies between pots due to manufacturing tolerance) is set to Pitch Bend 0?

(I hope that makes sense... I have never implemented Pitch Bend on a controller)

**Reply by paradajz** [reply]
It sends values 0-16383, it's up to your software how it will handle that. Lots of software handle middle value (8192) as 0, for instance, but YMMV.

**Comment by armandhammer** [comment]
Ok I guess I will hook up a pot to my ODv2 board and see what happens when I assign it pitch bend. I will be using it to control hardware synths, from old Korg DW8000 to new Novation Peak. So I am confused how I can calibrate a center detent pot so that the center resistance is “0” for pitch bend, and not erroneously sending a slight bend-up or bend-down when the pot is at the center click.

Other CC like Mod Wheel seem like no problem because 0 ohm is the same as 0 CC level.

**Reply by paradajz** [reply]
Let me know how it goes and then we can see if any improvements can be made. Pitch Bend is very lightly tested.

**Reply by armandhammer** [reply]
Cool, I will try it soon. Maybe today, definitely next week.

Is it possible for you to add an option for more MIDI DIN outs? It would be really useful. It seems [Badass MIDI Controller can do 4x](https://github.com/neroroxxx/BMC/blob/master/docs/midi-ports.md), but I'm thinking this might be because Teensy has so many serial ports available... and Arduino don't?

**Reply by paradajz** [reply]
What would be the use case for that? How you image it to work? Would it be simply two identical MIDI out ports with same traffic going on or something configurable? If the former, that could be added to the Mega since it has more serial ports, but if the former, that would be very non-trivial right now.

**Reply by armandhammer** [reply]
A duplicate port would be fine. I just want to route some controls to Synth 1 (e.g. Ch1) and other controls to Synth 2 (e.g. Ch2), without having to run them daisy chained with a THRU port (all my synths MIDI is individually connected to a TRS patch bay).
